src/main/java/com/monumental/services/MonumentService.java

```diff
@@ -154,19 +178,38 @@ private Predicate buildTagsQuery(CriteriaBuilder builder, CriteriaQuery query, R
      * @param distance - The distance from the comparison point to search in, units of miles
      * @param tags - List of tag names to search by
      * @param materials - List of material tag names to search by
+     * @param sortType - The way in which to sort the results by
      */
     private void buildSearchQuery(CriteriaBuilder builder, CriteriaQuery query, Root root, String searchQuery,
                                   Double latitude, Double longitude, Integer distance, List<String> tags,
-                                  List<String> materials, Boolean orderByResults) {
+                                  List<String> materials, SortType sortType) {
 
         List<Predicate> predicates = new ArrayList<>();
 
+        boolean sortByRelevance = false;
+        boolean sortByDistance = false;
+
+        switch (sortType) {
+            case NEWEST:
+                this.sortByDate(builder, query, root, true);
+                break;
+            case OLDEST:
+                this.sortByDate(builder, query, root, false);
+                break;
+            case DISTANCE:
+                sortByDistance = true;
+            case RELEVANCE:
+                sortByRelevance = true;
+            default:
+                break;
+        }
+
         if (!isNullOrEmpty(searchQuery)) {
-            predicates.add(this.buildSimilarityQuery(builder, query, root, searchQuery, 0.1, orderByResults));
+            predicates.add(this.buildSimilarityQuery(builder, query, root, searchQuery, 0.1, sortByRelevance));
         }
 
         if (latitude != null && longitude != null && distance != null) {
-            predicates.add(this.buildDWithinQuery(builder, root, latitude, longitude, distance));
+            predicates.add(this.buildDWithinQuery(builder, query, root, latitude, longitude, distance, sortByDistance));
```

Comment:
I'm a bit confused about what happens if a text AND location search happens and the `SortType` is `DISTANCE`. I **think** we'll first order the results by relevance, then distance? Is that right? And if so, is that what behavior we want?

Reply:
This shouldn't ever have 2 sorts happening. That *may* have been possible before because I was missing breaks in the switch statement, but now you should only ever have `sortByDistance` OR `sortByRelevance` not both.

Reply:
Alright yes, it was because you were missing `break`s. I thought at first that it might have been intentional, which is why I asked.